Specifications/QIR/Callables.md

```diff
@@ -318,6 +312,15 @@ language-specific compiler.
 The lambda body may need to include additional code that performs argument tuple
 construction before calling the underlying callable.
 
+#### Memory Management Table
+
+Since any captured values need to remain alive as long as the callable value exists, they also need to be unreferenced when the callable value is released. While sufficient type information for the captured values is known upon creation of the value, the information is no longer available at the time when it is released.
+Upon creation, a table with two function pointers for modifying reference and alias counts for captured values is hence associated with a callable value. 
+
+Like the implementation table, the table is defined as global constant with a unique name. It contains two pointers of type `void(%Tuple*, i64)*`; the first one points to the function for modifying the reference counts of captured values, the second points to the one for modifying the alias counts. Either of those pointers may be null, and if no values were captured, a null pointer should be passed instead of a table upon callable creation.
+
+Like for tuple and array items, the responsibility of managing the reference and access count for captured values lays with the compiler. The two functions can be invoked using the runtime function `__quantum__rt__callable_memory_management`, see the description [below](https://github.com/microsoft/qsharp-language/blob/main/Specifications/QIR/Callables.md#runtime-functions). 
```

Comment:
It might be worth giving an example where the ref count or alias count of the capture tuple elements could change without the ref count or access count of the callable changing. The simplest I can come up with is pretty trivial, in pseudo-Q#-ish:
`let f = lambda (x : Int[], y : Int) -> (x, y);`
Then every call to `f` has to increment the ref and access count of the capture tuple elements, specifically of the `x` array.

Reply:
I assume you meant to have x be captured in this example. This example wouldn't actually require going via the capture update function every time. the only time you really need to update the capture counts is when you update the callable counts. However, neither the compiler not the runtime has enough type information to do that unless we save those functions with the callable, and it just seems more consistent with how other counts work to have count changes always apply only to the exact value given as argument, and not anything "inside" that value.